packages/java-parser/api.d.ts

```diff
@@ -3498,8 +3498,6 @@ export interface PatternCstNode extends CstNode {
 
 export type PatternCtx = {
   primaryPattern: PrimaryPatternCstNode[];
-  AndAnd?: IToken[];
```

Comment:
I am not a fan of modifying the CST in order to make it easier to print the code afterwards, as it makes the java parser maintenance much more difficult for each deviation from the spec

Reply:
@clementdessoude I looked into the spec while implementing this solution, because I had similar concerns. What I found was that a [Pattern](https://docs.oracle.com/javase/specs/jls/se20/html/jls-14.html#jls-Pattern) is a [TypePattern](https://docs.oracle.com/javase/specs/jls/se20/html/jls-14.html#jls-TypePattern), which is a [LocalVariableDeclaration](https://docs.oracle.com/javase/specs/jls/se20/html/jls-14.html#jls-LocalVariableDeclaration), meaning that according to the spec, a Pattern should not contain AndAnd or a binaryExpression within it. I made this change because it seemed to make the parser more aligned with the spec.

All that being said, this solution is failing tests because it seems to have slowed down parsing, so I'll need to find a fix for that. But I do think this change to the CST is appropriate, unless I'm missing something.

Reply:
It seems that the `&&` version of the switch label "guarded" pattern is no more, and has been replaced with `when` in Java 21 (the first version that supports switch label pattern matching outside of a preview feature), which is scheduled to release tomorrow. So I will likely need to rethink this solution regardless. https://docs.oracle.com/javase/specs/jls/se21/html/jls-14.html#jls-SwitchLabel